src/sysupdate/sysupdate-transfer.h

```diff
@@ -0,0 +1,61 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <sys/types.h>
+
+#include "sd-id128.h"
+
+typedef struct Transfer Transfer;
```

Comment:
any reason the typedef is between includes?

Reply:
yes. The lower includes need the type to be defined. i.e. the header files include each other.

Reply:
(we do something similar between unit.h in the core: sometimes it makes sense to split up header files along object lines but still have them import each other)

Reply:
Shouldn't the typedef be in those headers then? Otherwise everytime they are included, it needs to be redefined

Reply:
not sure i follow? what do you mean by "redefined"?

Reply:
I mean if those headers need this forward declaration to work, shouldn't it be included in them directly, rather than in the place that uses them?

Reply:
hmm, dunno, so far we didn't do it that way. I could see why doing it this way is preferable, since it abstracts the precise definition (i.e. whether it is actually a struct or a union) away int to the header that later defines its contents too. If the forward decl would be placed in the consuming rather than the providing header you'd "leak" if this is a union or a struct.

Dunno. I'll grant you that this reason isn't a particular good one. It's just tat we did things this way so far not sure if there's a good reason to depart from that? And if we depart from it we probably should switch everything over, which I think should be a different PR.

Reply:
ok, that makes sense